pkg/k2/language_host/language_host.go

```diff
@@ -37,7 +48,12 @@ func (f *ServerAddress) Write(b []byte) (int, error) {
 		// address is the first match
 		f.Address = matches[1]
 		f.Log.Debugf("Found language host listening on %s", f.Address)
-		close(f.HasAddr)
+		close(f.Done)
+	}
+	matches = exceptionPattern.FindStringSubmatch(s)
+	if len(matches) >= 2 {
+		f.Error = errors.New(strings.TrimSpace(matches[1]))
+		close(f.Done)
```

Comment:
if the address is found, then it errors (or it errors twice), this will double close the channel which IIRC will panic. Make sure to check for `Address` or `Error` before closing